dom/base/nsJSUtils.h

```diff
@@ -119,12 +119,14 @@ inline bool AssignJSString(JSContext* cx, T& dest, JSString* s) {
     if (chars[len] == '\0') {
       AssignFromStringBuffer(
           nsStringBuffer::FromData(const_cast<char16_t*>(chars)), len, dest);
+      dest.AssignTaint(JS_GetStringTaint(s));
```

Comment:
It looks like this functionality has moved in the main branch. I think this change needs to be moved into https://github.com/SAP/project-foxhound/blob/1975ec03924276705b0dfddbfa173bb6fea0b6bb/js/xpconnect/src/xpcpublic.h#L432

Reply:
Yes, I noticed that the code moved. I have updated it and added the code in the correct place. I tested it and it works fine but I am still having issues running the mochitests.

Reply:
Great, thanks for the update!